From 6857b66383bc634f3cf6f3fffa8da38e317ef478 Mon Sep 17 00:00:00 2001
From: Ralph Bean <rbean@redhat.com>
Date: Mon, 29 Oct 2012 10:36:33 -0400
Subject: [PATCH] Disable epgm test

---
 txzmq/test/test_pubsub.py | 42 ------------------------------------------
 1 file changed, 42 deletions(-)

diff --git a/txzmq/test/test_pubsub.py b/txzmq/test/test_pubsub.py
index 04ef4f0..b118a59 100644
--- a/txzmq/test/test_pubsub.py
+++ b/txzmq/test/test_pubsub.py
@@ -17,23 +17,6 @@ class ZmqTestSubConnection(ZmqSubConnection):
         self.messages.append([tag, message])
 
 
-def _detect_epgm():
-    """
-    Utility function to test for presence of epgm:// in zeromq.
-    """
-    import zmq
-
-    context = zmq.Context()
-    socket = zmq.Socket(context, zmq.core.constants.PUB)
-
-    try:
-        socket.bind("epgm://127.0.0.1;239.192.1.1:5557")
-
-        return True
-    except zmq.core.error.ZMQError:
-        return False
-
-
 class ZmqConnectionTestCase(unittest.TestCase):
     """
     Test case for L{zmq.twisted.connection.Connection}.
@@ -79,28 +62,6 @@ class ZmqConnectionTestCase(unittest.TestCase):
         return _wait(0.01).addCallback(publish) \
             .addCallback(lambda _: _wait(0.01)).addCallback(check)
 
-    def test_send_recv_pgm(self):
-        r = ZmqTestSubConnection(self.factory, ZmqEndpoint(
-            ZmqEndpointType.bind, "epgm://127.0.0.1;239.192.1.1:5556"))
-
-        s = ZmqPubConnection(self.factory, ZmqEndpoint(
-            ZmqEndpointType.connect, "epgm://127.0.0.1;239.192.1.1:5556"))
-
-        r.subscribe('tag')
-
-        def publish(ignore):
-            s.publish('xyz', 'different-tag')
-            s.publish('abcd', 'tag1')
-
-        def check(ignore):
-            result = getattr(r, 'messages', [])
-            expected = [['tag1', 'abcd']]
-            self.failUnlessEqual(
-                result, expected, "Message should have been received")
-
-        return _wait(0.2).addCallback(publish) \
-            .addCallback(lambda _: _wait(0.2)).addCallback(check)
-
     def test_send_recv_multiple_endpoints(self):
         r = ZmqTestSubConnection(
             self.factory,
@@ -128,6 +89,3 @@ class ZmqConnectionTestCase(unittest.TestCase):
 
         return _wait(0.1).addCallback(publish) \
             .addCallback(lambda _: _wait(0.1)).addCallback(check)
-
-    if not _detect_epgm():
-        test_send_recv_pgm.skip = "epgm:// not available"
-- 
1.7.12.1

