From 7507ce3f5fc2646ba8fe6be67e523a51b26e1706 Mon Sep 17 00:00:00 2001
From: Ralph Bean <rbean@redhat.com>
Date: Mon, 21 May 2012 18:18:35 -0400
Subject: [PATCH] Disable EPGM test

---
 txzmq/test/test_pubsub.py |   18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/txzmq/test/test_pubsub.py b/txzmq/test/test_pubsub.py
index 000376d..cd64110 100644
--- a/txzmq/test/test_pubsub.py
+++ b/txzmq/test/test_pubsub.py
@@ -50,24 +50,6 @@ class ZmqConnectionTestCase(unittest.TestCase):
 
         return _wait(0.01).addCallback(check)
 
-    def test_send_recv_pgm(self):
-        r = ZmqTestSubConnection(self.factory, ZmqEndpoint(
-            ZmqEndpointType.bind, "epgm://127.0.0.1;239.192.1.1:5556"))
-        s = ZmqPubConnection(self.factory, ZmqEndpoint(
-            ZmqEndpointType.connect, "epgm://127.0.0.1;239.192.1.1:5556"))
-
-        r.subscribe('tag')
-        s.publish('xyz', 'different-tag')
-        s.publish('abcd', 'tag1')
-
-        def check(ignore):
-            result = getattr(r, 'messages', [])
-            expected = [['tag1', 'abcd']]
-            self.failUnlessEqual(
-                result, expected, "Message should have been received")
-
-        return _wait(0.2).addCallback(check)
-
     def test_send_recv_multiple_endpoints(self):
         r = ZmqTestSubConnection(
             self.factory,
-- 
1.7.10.2

