From 0be4152e7ed7aaa841d293cbab252babc5df11a3 Mon Sep 17 00:00:00 2001
From: Ralph Bean <rbean@redhat.com>
Date: Tue, 26 Mar 2013 23:17:34 -0400
Subject: [PATCH 5/6] Double compat checking (ridiculous!)

---
 txzmq/connection.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/txzmq/connection.py b/txzmq/connection.py
index fbfb1be..1d73d32 100644
--- a/txzmq/connection.py
+++ b/txzmq/connection.py
@@ -18,7 +18,12 @@ try:
 
     ZMQ3 = int(version.zmq_version().split('.')[0]) >= 3
 except ImportError:
-    ZMQ3 = False
+    try:
+        # In pyzmq-13.0.0, this moved again.
+        from zmq.core import zmq_version_info
+        ZMQ3 = zmq_version_info()[0] >= 3
+    except ImportError:
+        ZMQ3 = False
 
 
 class ZmqEndpointType(object):
-- 
1.8.1.4

