From 54d6d163a478d1d393d058c73eb9731661cb6eee Mon Sep 17 00:00:00 2001
From: Ralph Bean <rbean@redhat.com>
Date: Wed, 10 Oct 2012 14:46:07 -0400
Subject: [PATCH] Support older pyzmq.

---
 txzmq/connection.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/txzmq/connection.py b/txzmq/connection.py
index 815c47d..38af495 100644
--- a/txzmq/connection.py
+++ b/txzmq/connection.py
@@ -13,7 +13,7 @@ from twisted.python import log
 
 
 # Patch zmq.core.constants to support both zeromq2 and zeromq3
-ZMQ3 = version.zmq_version_info()[0] >= 3
+ZMQ3 = int(version.zmq_version().split('.')[0]) >= 3
 
 if not ZMQ3:
     constants.DONTWAIT = constants.NOBLOCK
-- 
1.7.12.1

